code/game/objects/gamemode/crash/machines.dm

```diff
@@ -0,0 +1,160 @@
+	. = ..()
+
+	if(!disc_type)
+		stack_trace("disc_type is required to be set before init")
+		return INITIALIZE_HINT_QDEL
+
+/obj/machinery/computer/nuke_disc_generator/process()
+	. = ..()
+	if(. || !current_timer)
+		return
+
+	deltimer(current_timer)
+	visible_message("<b>[src]</b> shuts down as it loses power. Any running programs will now exit")
+
+/obj/machinery/computer/nuke_disc_generator/attack_hand(mob/user)
+	. = ..()
+	if(.)
+		return
+
+	interact(user)
+
+/obj/machinery/computer/nuke_disc_generator/interact(mob/user)
+	user.set_interaction(src)
+	var/dat = ""
+	dat += "<a href='?src=[REF(src)];mach_close=computer'>Close</a><br><br>"
+	dat += "<div align='center'><a href='?src=[REF(src)];generate=1'>Run Program</a></div>"
+	dat += "<br/>"
+	dat += "<hr/>"
+	dat += "<div align='center'><h2>Status</h2></div>"
+
+	var/message = "Error"
+	if(completed_segments >= total_segments)
+		message = "Disc generated. Run program to print."
+	else if(current_timer)
+		message = "Program running"
+	else if(completed_segments == 0)
+		message = "Idle"
+	else if(completed_segments < total_segments)
+		message = "Restart required. Please rerun the program"
+	else
+		message = "Unknown"
+
+	var/progress = round((completed_segments / total_segments) * 100)
+
+	dat += "<br/><span><b>Progress</b>: [progress]%</span>"
+	dat += "<br/><span><b>Time left</b>: [current_timer ? round(timeleft(current_timer) * 0.1, 2) : 0.0]</span>"
+	dat += "<br/><span><b>Message</b>: [message]</span>"
+
+	var/flair = ""
+	if(completed_segments > 0)
+		for(var/i in 1 to completed_segments)
+			flair += "[technobabble[i]]<br />"
+
+		dat += "<br /><br /><span style='font-family: monospace, monospace;'>[flair]</span>"
+
+	var/datum/browser/popup = new(user, "computer", "<div align='center'>Nuke Disc Generator</div>")
+	popup.set_content(dat)
+	popup.open()
+
+/obj/machinery/computer/nuke_disc_generator/Topic(href, href_list)
+	. = ..()
+	if(.)
+		return
+
+	if(href_list["generate"])
+		if(printing || current_timer)
+			to_chat(usr, "<span class='warning'>A program is already running.</span>")
+			return
+		if(reprintable)
+			printing = TRUE
+			addtimer(VARSET_CALLBACK(src, printing, FALSE), printing_time) // TODO Change to larger time
+
+			usr.visible_message("[usr] started a program to regenerate a nuclear disc code.", "You started a program to generate a nuclear disc code.")
+			if(!do_after(usr, printing_time, TRUE, src, BUSY_ICON_GENERIC, null, null, CALLBACK(src, /datum.proc/process)))
+				return
+
+			print_disc()
+			return
+
+		printing = TRUE
+		addtimer(VARSET_CALLBACK(src, printing, FALSE), segment_time) // TODO Change to larger time
+
+		usr.visible_message("[usr] started a program to generate a nuclear disc code.", "You started a program to generate a nuclear disc code.")
+		if(!do_after(usr, segment_time, TRUE, src, BUSY_ICON_GENERIC, null, null, CALLBACK(src, /datum.proc/process)))
+			current_timer = null
```

Comment:
deltimer?

Reply:
This was incorrectly left over, its gonna be removed